tests/FakeItEasy.Specs/CancellationSpecs.cs

```diff
@@ -149,7 +157,7 @@ public void AsyncNonCancelledCancellationToken(IFoo fake, CancellationToken canc
                 .x(() => cancellationToken = new CancellationToken(false));
 
             "When an async method is called with this cancellation token"
-                .x(() => task = fake.BarAsync(cancellationToken));
+                .x(() => { task = fake.BarAsync(cancellationToken); });

```

Comment:
Ah, yes, I intended to comment on it but I forgot. The `x` method has 2 overloads (4 actually, but we rarely use the ones with an `IStepContext`):
- one that takes an `Action`
- and one that takes a `Func<Task>`, for async scenarios

When I use

    .x(() => task = fake.BarAsync(cancellationToken))

the expression returns a task, so it's inferred to be a `Func<Task>`. And in that case, xBehave awaits the task. This is not what we want here (we just want to assign the task, not return it), so I use a statement body to force it to be inferred as `Action`. I had already noticed the problem for cancelled tasks because it caused the steps to fail, but I thought it would be more consistent to do the same for successful tasks.